### ISSUE TITLE
Module Build Error

### ISSUE THREAD

**Issue by saulshanabrook** [deepsweet/autopolyfiller-loader#12]
I have just installed and set up this loader, and when I try to build I get an error:

```bash
$ npm run dev-server                                                                                                                                                                                                             ≣≣≣≣≣≣≣≣≣  ≣

> caido@0.1.0 dev-server /Users/saul/projects/go/src/github.com/lucibus/lucibus/caido
> env WEBPACK_DEV_SERVER=true WEBPACK_DEVTOOL=eval-source-map WEBPACK_DEBUG=true webpack-dev-server --colors --progress --hot

 70% 1/1 build moduleshttp://0.0.0.0:8080/
webpack result is served from /
content is served from /Users/saul/projects/go/src/github.com/lucibus/lucibus/caido
Hash: 62701bec273b5fb3fb07
Version: webpack 1.11.0
Time: 4431ms
chunk    {0} main.js (main) 173 kB [rendered]
    [0] multi main 52 bytes {0} [built] [1 error]
  [157] ./~/webpack-dev-server/client?http://0.0.0.0:8080 1.82 kB {0} [built]
  [158] ./~/webpack-dev-server/~/socket.io-client/index.js 37 bytes {0} [built]
  [159] ./~/webpack-dev-server/~/socket.io-client/lib/index.js 1.5 kB {0} [built]
  [160] ./~/webpack-dev-server/~/socket.io-client/lib/url.js 1.53 kB {0} [built]
  [161] ./~/webpack-dev-server/~/socket.io-client/~/parseuri/index.js 690 bytes {0} [built]
  [162] ./~/webpack-dev-server/~/socket.io-client/~/debug/debug.js 2.46 kB {0} [built]
  [163] ./~/webpack-dev-server/~/socket.io-client/~/socket.io-parser/index.js 7.63 kB {0} [built]
  [164] ./~/webpack-dev-server/~/socket.io-client/~/socket.io-parser/~/json3/lib/json3.js 40.1 kB {0} [built]
  [165] (webpack)/buildin/amd-options.js 43 bytes {0} [built]
  [166] ./~/webpack-dev-server/~/socket.io-client/~/socket.io-parser/~/isarray/index.js 120 bytes {0} [built]
  [167] ./~/webpack-dev-server/~/socket.io-client/~/component-emitter/index.js 3 kB {0} [built]
  [168] ./~/webpack-dev-server/~/socket.io-client/~/socket.io-parser/binary.js 3.84 kB {0} [built]
  [169] ./~/webpack-dev-server/~/socket.io-client/~/socket.io-parser/is-buffer.js 252 bytes {0} [built]
  [170] ./~/webpack-dev-server/~/socket.io-client/lib/manager.js 11 kB {0} [built]
  [171] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/index.js 38 bytes {0} [built]
  [172] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/lib/index.js 140 bytes {0} [built]
  [173] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/lib/socket.js 16.9 kB {0} [built]
  [174] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/lib/transports/index.js 1.08 kB {0} [built]
  [175] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/lib/xmlhttprequest.js 1.16 kB {0} [built]
  [176] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/has-cors/index.js 458 bytes {0} [built]
  [177] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/has-cors/~/global/index.js 257 bytes {0} [built]
  [178] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/lib/transports/polling-xhr.js 7.92 kB {0} [built]
  [179] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/lib/transports/polling.js 4.85 kB {0} [built]
  [180] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/lib/transport.js 2.74 kB {0} [built]
  [181] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/engine.io-parser/lib/browser.js 14.4 kB {0} [built]
  [182] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/engine.io-parser/lib/keys.js 297 bytes {0} [built]
  [183] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/engine.io-parser/~/has-binary/index.js 1.06 kB {0} [built]
  [184] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/engine.io-parser/~/has-binary/~/isarray/index.js 120 bytes {0} [built]
  [185] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/engine.io-parser/~/arraybuffer.slice/index.js 725 bytes {0} [built]
  [186] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/engine.io-parser/~/base64-arraybuffer/lib/base64-arraybuffer.js 1.7 kB {0} [built]
  [187] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/engine.io-parser/~/after/index.js 685 bytes {0} [built]
  [188] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/engine.io-parser/~/utf8/utf8.js 6.22 kB {0} [built]
  [189] (webpack)/buildin/module.js 251 bytes {0} [built]
  [190] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/engine.io-parser/~/blob/index.js 973 bytes {0} [built]
  [191] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/parseqs/index.js 708 bytes {0} [built]
  [192] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/component-inherit/index.js 146 bytes {0} [built]
  [193] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/debug/browser.js 3.28 kB {0} [built]
  [194] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/debug/debug.js 4.1 kB {0} [built]
  [195] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/debug/~/ms/index.js 2.02 kB {0} [built]
  [196] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/lib/transports/polling-jsonp.js 4.82 kB {0} [built]
  [197] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/lib/transports/websocket.js 4.57 kB {0} [built]
  [198] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/ws/lib/browser.js 831 bytes {0} [built]
  [199] ./~/webpack-dev-server/~/socket.io-client/~/indexof/index.js 199 bytes {0} [built]
  [200] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/parseuri/index.js 1.19 kB {0} [built]
  [201] ./~/webpack-dev-server/~/socket.io-client/~/engine.io-client/~/parsejson/index.js 842 bytes {0} [built]
  [202] ./~/webpack-dev-server/~/socket.io-client/lib/socket.js 6.99 kB {0} [built]
  [203] ./~/webpack-dev-server/~/socket.io-client/~/to-array/index.js 216 bytes {0} [built]
  [204] ./~/webpack-dev-server/~/socket.io-client/lib/on.js 370 bytes {0} [built]
  [205] ./~/webpack-dev-server/~/socket.io-client/~/component-bind/index.js 474 bytes {0} [built]
  [206] ./~/webpack-dev-server/~/socket.io-client/~/has-binary/index.js 1.08 kB {0} [built]
  [207] ./~/webpack-dev-server/~/socket.io-client/~/has-binary/~/isarray/index.js 120 bytes {0} [built]
  [208] ./~/webpack-dev-server/~/socket.io-client/~/object-component/index.js 1.18 kB {0} [built]
  [209] ./~/webpack-dev-server/~/socket.io-client/~/backo2/index.js 1.4 kB {0} [built]
  [210] ./~/webpack-dev-server/~/strip-ansi/index.js 161 bytes {0} [built]
  [211] ./~/webpack-dev-server/~/strip-ansi/~/ansi-regex/index.js 145 bytes {0} [built]
  [212] ./~/webpack/hot/dev-server.js 1.85 kB {0} [built]
  [213] ./~/webpack/hot/log-apply-result.js 813 bytes {0} [built]

ERROR in ./app/main.jsx
Module build failed: TypeError: Function.prototype.apply: Arguments list has wrong type
    at Object.module.exports (/Users/saul/projects/go/src/github.com/lucibus/lucibus/caido/node_modules/autopolyfiller-loader/index.js:12:36)
 @ multi main
webpack: bundle is now VALID.
```

I can't tell if this is an error in `autopolyfiller-loader` or my source code.

My `./app/main.jsx` file looks like this:

```javascript
import React from 'react'

import App from './containers/App'
import controller from './controller'
import Storage from './Storage'

import addSystem from './actions/addSystem.js'
import createAddressOneSystem from './actions/createAddressOneSystem.js'
import moveSystem from './actions/moveSystem.js'
import outputSynced from './actions/outputSynced.js'
import selectSystem from './actions/selectSystem.js'
import setSynced from './actions/setSynced.js'
import changeSystemLevel from './actions/changeSystemLevel.js'

// ACTIONS
controller.signal('gotWebsocketMessage', setSynced)
controller.signal('clickedAddAddressOne', createAddressOneSystem, addSystem, [outputSynced])
controller.signal('draggedSystem', moveSystem, [outputSynced])
controller.signal('clickedSystem', selectSystem)
controller.signal('movedInspector', changeSystemLevel, [outputSynced])

Storage.onMessage(controller.signals.gotWebsocketMessage)

React.render(
  controller.injectInto(App),
  document.getElementById('content')
)
```

I can't find any uses of `apply` in my source code. I am, however, using `babel-loader` before this loader, so maybe that is making some use of apply that is wrong? My js loaders look like:

```javascript
['autopolyfiller?browsers=["last 2 versions", "ie >= 9" ]', 'babel-loader?stage=0']
```

**Comment by saulshanabrook** [comment]
OK I was enabling the loader wrong (using `loader` instead of `loaders`). Now when I start, I get a different error:


```
ERROR in ./app/Storage.js
Module build failed: Error: Cannot find module 'polyfill-service/polyfills/Window.prototype.JSON/polyfill'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.require.resolve (module.js:388:19)
    at /Users/saul/projects/go/src/github.com/lucibus/lucibus/caido/node_modules/autopolyfiller-loader/index.js:23:59
    at Array.forEach (native)
    at Object.module.exports (/Users/saul/projects/go/src/github.com/lucibus/lucibus/caido/node_modules/autopolyfiller-loader/index.js:22:19)
 @ ./app/main.jsx 19:15-35
```

**Comment by saulshanabrook** [comment]
I tried fixing that by `npm install --save polyfill-service`, but I still got the same error.

**Comment by deepsweet** [comment]
it looks like [autopolyfiller](https://github.com/azproduction/autopolyfiller) provides wrong polyfill name from view of the latest [polyfill-service](https://github.com/financial-times/polyfill-service) – there is no `polyfill-service/polyfills/Window.prototype.JSON/polyfill` file.

@azproduction is there any news or updates? I didn't use it for a very long time, so I just have no idea what is going on :)

**Comment by azproduction** [comment]
@deepsweet there was a lot of back and forths with polyfill-service at the end I was tier of waiting. Now I have different priorities in OSS, so autopolyfiller is a bit abandoned now.

@saulshanabrook autopolyfiller now is using the fork of polyfill-service which is far before the origin – https://github.com/azproduction/polyfill/tree/master/source
So there is no need to upgrade, because they are just incompatible. Can you also try 
`autopolyfiller?{"browsers":["last 2 versions", "ie >= 9"]}` ?

**Comment by saulshanabrook** [comment]
@azproduction Tried that same error.

**Comment by toddpi314** [comment]
The polyfill is there. It just looks like the naming convention used for module lookup needs to exclude the `Window.prototype.`

**Comment by toddpi314** [comment]
@azproduction Can you please verify https://github.com/deepsweet/autopolyfiller-loader/pull/14 and get an update out to NPM?

**Comment by deepsweet** [comment]
https://github.com/deepsweet/autopolyfiller-loader/issues/17

**Comment by NumminorihSF** [comment]
Fixed on #14
